frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -188,6 +228,85 @@ const getKeysToExpand = (preExpandPanelKey, tableKey) => {
   return keysToExpand;
 };
 
+const getFormattedChildrenData = (item, formatChildrenData) =>
+  item.typeMetadata
+    ? [item.typeMetadata].map(formatChildrenData)
+    : item.children.map(formatChildrenData);
+
+const getSpecificTypeOpenerRow = (
+  colToDisplay,
+  arrayCount,
+  nestedLevel,
+  additionalTableColCount
+) => {
+  let arrayOpenerLabel = '';
+  if (arrayCount > 0) {
+    arrayOpenerLabel = ARRAY_LABEL + ARRAY_OPENER.repeat(arrayCount);
+  }
+
+  let mapOpenerLabel = '';
+  if (colToDisplay.kind === MAP_KIND) {
+    mapOpenerLabel = MAP_LABEL + MAP_OPENER;
+  }
+
+  const cellStyle = {
+    paddingLeft: getIndentationPaddingSize(nestedLevel, false),
+  };
+
+  return (
+    <tr
+      className="ams-table-row is-nested-column-row is-specific-type-row"
+      key={`opener_row:${colToDisplay.key}`}
+    >
+      <td key={`opener_cell:${colToDisplay.key}`} style={cellStyle}>
+        <span className="column-type-label">
+          {arrayOpenerLabel} {mapOpenerLabel}
+        </span>
+      </td>
+      {[...Array(additionalTableColCount)].map((value, index) => (
```

Comment:
This is the way I came across when looking for a how to add a dynamic number of elements. It does add the correct number of empty `<td>`s by iterating over the count, and the reason I'm doing this is bc otherwise the background-color was being cut off at the width of the first table column td